azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -591,9 +597,15 @@ private void finalizeFlow(final ExecutableFlowBase flow) {
     final long durationSec = (flow.getEndTime() - flow.getStartTime()) / 1000;
     switch (flow.getStatus()) {
       case FAILED_FINISHING:
-        this.logger.info("Setting flow '" + id + "' status to FAILED in "
+        final Status status = isKilled() ? Status.KILLED : Status.FAILED;
```

Comment:
Sorry, that was a mistake. There's something more that's needed because of KILLING status, but this is not the correct way. As you can see some assertions had been also changed in the FlowRunnerTest2, but I hadn't really thought about those. I should have been more careful.